### PR TITLE
Adding -H + API token to curl commands

### LINKED ISSUE #7993: Native API: Certain curl commands require -H + API token
Otherwise, the native API rejects the user's requests on the ground that they are a 'guest'.

This concerns 4 commands listed in the [Get JSON Representation of a Dataset](https://guides.dataverse.org/en/latest/api/native-api.html#get-json-representation-of-a-dataset) section of the guide.

### PR DESCRIPTION
**Pull request by BPeuch**
**What this PR does / why we need it**: Without ``-H`` and the API token in these curl commands, the native API rejects the user's requests on the ground that they are a 'guest'.

**Which issue(s) this PR closes**:

Closes #7993

**Additional documentation**: Thanks to Dataverse chat user Lincoln for helping me out with this one!

### CODE REVIEW

File: doc/sphinx-guides/source/api/native-api.rst [hunk 1]
```diff
@@ -609,13 +609,13 @@ Example: Getting the dataset whose DOI is *10.5072/FK2/J8SJZB*:
   export SERVER_URL=https://demo.dataverse.org
   export PERSISTENT_IDENTIFIER=doi:10.5072/FK2/J8SJZB
 
-  curl $SERVER_URL/api/datasets/:persistentId/?persistentId=$PERSISTENT_IDENTIFIER
+  curl -H "X-Dataverse-key:$API_KEY" $SERVER_URL/api/datasets/:persistentId/?persistentId=$PERSISTENT_IDENTIFIER
```
Comment (pdurbin) [inline doc/sphinx-guides/source/api/native-api.rst:612]:
```suggestion
  curl -H "X-Dataverse-key:$API_TOKEN" $SERVER_URL/api/datasets/:persistentId/?persistentId=$PERSISTENT_IDENTIFIER
```

File: doc/sphinx-guides/source/api/native-api.rst [hunk 2]
```diff
@@ -609,13 +609,13 @@ Example: Getting the dataset whose DOI is *10.5072/FK2/J8SJZB*:
...
 
 The fully expanded example above (without environment variables) looks like this:
 
 .. code-block:: bash
 
-  curl https://demo.dataverse.org/api/datasets/:persistentId/?persistentId=doi:10.5072/FK2/J8SJZB
+  curl -H "X-Dataverse-key:$API_KEY" https://demo.dataverse.org/api/datasets/:persistentId/?persistentId=doi:10.5072/FK2/J8SJZB
```
Comment (pdurbin) [inline doc/sphinx-guides/source/api/native-api.rst:618]:
```suggestion
  curl -H "X-Dataverse-key:$API_TOKEN" https://demo.dataverse.org/api/datasets/:persistentId/?persistentId=doi:10.5072/FK2/J8SJZB
```

File: doc/sphinx-guides/source/api/native-api.rst [hunk 3]
```diff
@@ -624,13 +624,13 @@ Getting its draft version:
   export SERVER_URL=https://demo.dataverse.org
   export PERSISTENT_IDENTIFIER=doi:10.5072/FK2/J8SJZB
 
-  curl http://$SERVER/api/datasets/:persistentId/versions/:draft?persistentId=$PERSISTENT_IDENTIFIER
+  curl -H "X-Dataverse-key:$API_KEY" http://$SERVER/api/datasets/:persistentId/versions/:draft?persistentId=$PERSISTENT_IDENTIFIER
```
Comment (pdurbin) [inline doc/sphinx-guides/source/api/native-api.rst:627]:
```suggestion
  curl -H "X-Dataverse-key:$API_TOKEN" http://$SERVER/api/datasets/:persistentId/versions/:draft?persistentId=$PERSISTENT_IDENTIFIER
```

File: doc/sphinx-guides/source/api/native-api.rst [hunk 4]
```diff
@@ -624,13 +624,13 @@ Getting its draft version:
...
 
 The fully expanded example above (without environment variables) looks like this:
 
 .. code-block:: bash
 
-  curl https://demo.dataverse.org/api/datasets/:persistentId/versions/:draft?persistentId=doi:10.5072/FK2/J8SJZB
+  curl -H "X-Dataverse-key:$API_KEY" https://demo.dataverse.org/api/datasets/:persistentId/versions/:draft?persistentId=doi:10.5072/FK2/J8SJZB
```
Comment (pdurbin) [inline doc/sphinx-guides/source/api/native-api.rst:633]:
```suggestion
  curl -H "X-Dataverse-key:$API_TOKEN" https://demo.dataverse.org/api/datasets/:persistentId/versions/:draft?persistentId=doi:10.5072/FK2/J8SJZB
```

**Review by pdurbin — CHANGES_REQUESTED** [review]
@BPeuch thanks for the pull request! It looks great but can you please change these from `$API_KEY` to `$API_TOKEN`? This is for consistency with the rest of the page (and the guide).

**Comment by BPeuch** [comment]
Thanks for the feedback, Philip!
This is my first time reviewing suggestions here. I think I managed to accept three of them but bungled the fourth one. From where you are, has 'KEY' been replaced with 'TOKEN' in all four occurrences? 🤔

**Review by pdurbin — APPROVED** [review]
Looks good!

**Comment by pdurbin** [comment]
@BPeuch yes, all set for all four. Thanks!

**Comment by BPeuch** [comment]
Wonderful! Thanks a lot @pdurbin

**Pull request merged**